cdap-hbase-compat-base/src/main/java/co/cask/cdap/replication/TableUpdater.java

```diff
@@ -69,7 +69,7 @@ public TableUpdater(String rowType, final Configuration conf) {
    * @param time is the value for the Map
    */
   public void updateTime(String regionName, long time) throws IOException {
-    synchronized (cachedUpdates) {
+    synchronized (this) {
```

Comment:
this is functionally equivalent to making the method synchronized. In general, I think it's better to synchronize on a  `private final Object lock`. Otherwise you may unintentionally block if you add another synchronized method, or somebody happens to synchronize on this instance somewhere else for some reason.